chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -78,14 +88,50 @@ def __exit__(self, type, value, traceback):
             self.unlock_pool()
 
     def lock_pool(self):
-        lock = self.import_locks[self.pool_path]
-        lock.acquire()
-        # log.debug('ZfsDevice acquired lock on %s' % self.pool_path)
+        """
+        Attempt to acquire a lock on a given zpool through a lockfile, increment the reference count each time
+        this method is called, regardless of whether acquire is actually called. Keep trying until we know
+        we are locking the zpool.
+
+        Lock acquire polls the timeout at intervals of LOCK_ACQUIRE_TIMEOUT.
+        """
+        if ZfsDevice.locks_dir_initialized == False:
+            # ensure lock directory exists
+            try:
+                os.mkdir(self.ZPOOL_LOCK_DIR)
+            except OSError:
+                pass
+
+            ZfsDevice.locks_dir_initialized = True
+
+        while not self.lock.i_am_locking():
+            try:
+                self.lock.acquire(self.LOCK_ACQUIRE_TIMEOUT)
+                with open(self.lock.lock_file, 'w') as f:
+                    f.writelines(str(self.lock.pid))
+            except LockTimeout:
+                # prune locks owned by non-existent processes
+                with open(self.lock.lock_file, 'r') as f:
+                    contents = f.readlines()
+
+                assert len(contents) == 1 and contents[0].isdigit(), \
```

Comment:
I'm not sure there is one, if lockfiles are created and they don't contain the expected contents, the daemon should not continue as it would jeopardise pacemaker and HA operation